File: src/C#/BehavioralPatterns/Mediator/ChainReaction/Button.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ChainReaction
{
    public class Button : Participant
    {
        public Button(Mediator mediator) : base(mediator)
        {

        }

        public void Enable() => Console.WriteLine("Button is enabled.");

        public void Disable() => Console.WriteLine("Button is disabled.");
    }
}

File: src/JavaScript/BehavioralPatterns/Mediator/ChainReaction/button.js

const Participant = require("./participant");

class Button extends Participant {
    constructor(mediator) {
        super(mediator);
    }

    enable() {
        console.log("Button is enabled.");
    }

    disable() {
        console.log("Button is disabled.");
    }
}

module.exports = Button;